### PR TITLE
fixes for build samples by watcom

### PR DESCRIPTION
**Pull request by jmalak**
add DLL related build make file for watcom

### CODE REVIEW

**Comment by jmalak** [comment]
Please, update OW2.0 pack to include fixed nt/nldef.h.

**Comment by gvanem** [comment]
It is in the `watcom20.zip`. Should get unpacked to  `CI-temp/h/nt/nldef.h`.
Try it locally: 
```
cd %WATT_ROOT% 
set BUILDER=watcom 
set CPU=x86 
set MODEL=win32 
cmd /c appveoyr-script build_src
```
But edit to use this URL `set URL_WATCOM_ZIP=http://www.watt-32.net/CI/watcom20.zip`

Some issue with OneDrive again.

**Comment by gvanem** [comment]
Maybe this is better: https://github.com/gvanem/Watt-32/commit/2abce2a931dc4395705059d5cad84e0aa4c68860

File: src/winadinf.c [hunk 1]
```diff
@@ -1348,6 +1348,10 @@ static const struct search_list arp_types[] = {
                   { MIB_IPNET_TYPE_STATIC,  "Static" }
                 };
 
+#if defined(__WATCOMC__) && (__WATCOMC__ < 1300)
+  #define NlnsStale  NlnsState  /* typo in Watcom's <ntdef.h> */
```
Comment (gvanem) [inline src/winadinf.c:1352]: This issue was resolved. Please rebase.

File: bin/wcwindll.mak [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+.c.exe: .PRECIOUS
+      $(CC) $(CFLAGS) $[@
+      $(LINK) $(LFLAGS) name $^@ file $[&.obj
+
+clean: .SYMBOLIC
+      -@del *.obj
+      -@del *.map
+      -@del $(PROGS)
+
```
Comment (gvanem) [inline bin/wcwindll.mak:48]: All those `wmake` makefiles have a lot in common. So maybe they could include a `ow-com.mak` file?

**Comment by jmalak** [comment]
now wccommon.mak file contains common definition and rules for build samples by watcom
in this make file is definition of samples to build for each memory model that it must be updated if new sample is added

**Comment by gvanem** [comment]
Merging this large PR. Thanks again!

File: bin/wc_win.mak [hunk 3]
```diff
@@ -6,46 +6,12 @@
 #  Usage: 'wmake -h -f wc_win.mak'
 #
 
-.EXTENSIONS:
-.EXTENSIONS: .exe .obj .c
+MODEL   = win32
+CC      = *wcc386 -3r
+CFLAGS  = -bt=nt -mf -oaxt -DWIN32 -DWATT32_STATIC
```
Comment (gvanem) [inline bin/wc_win.mak:11]: Why are Win32 sample not using `watt-32.dll`?

Comment (jmalak) [reply bin/wc_win.mak:11]: It was first atempt to do working makefiles.
As I see you extended wc_win.mak to switch to use static or dynamic library version for compilation.

Comment (gvanem) [reply bin/wc_win.mak:11]: Some issue with Watcom `tracert.exe` when linked to `watt-32.dll`. `_arp_resolve` not found !?

**Pull request merged**